src/templates/prepared-map/usecase-detail/example/example-tree/parse.ts

```diff
@@ -0,0 +1,92 @@
+import type {
+  ComlinkListLiteralNode,
+  ComlinkLiteralNode,
+  ComlinkObjectLiteralNode,
+  ComlinkPrimitiveLiteralNode,
+} from '@superfaceai/ast';
+
+import type {
+  ExampleArray,
+  ExampleObject,
+  ExampleScalar,
+  UseCaseExample,
+} from '../usecase-example';
+
+export function parseObjectLiteral(
+  node: ComlinkObjectLiteralNode
+): ExampleObject {
+  const properties: ({ name: string } & (
+    | ExampleArray
+    | ExampleScalar
+    | ExampleObject
+  ))[] = [];
+  for (const field of node.fields) {
+    if (field.value.kind === 'ComlinkPrimitiveLiteral') {
+      properties.push({
+        name: field.key.join('.'),
+        ...parsePrimitiveLiteral(field.value),
+      });
+    } else if (field.value.kind === 'ComlinkListLiteral') {
+      properties.push({
+        name: field.key.join('.'),
+        ...parseListLiteral(field.value),
+      });
+    } else {
+      properties.push({
+        name: field.key.join('.'),
+        ...parseObjectLiteral(field.value),
+      });
+    }
+  }
+
+  return {
+    kind: 'object',
+    properties,
+  };
+}
+
+export function parseListLiteral(node: ComlinkListLiteralNode): ExampleArray {
+  const items: (ExampleArray | ExampleObject | ExampleScalar)[] = [];
+
+  for (const item of node.items) {
+    if (item.kind === 'ComlinkPrimitiveLiteral') {
+      items.push(parsePrimitiveLiteral(item));
+    } else if (item.kind === 'ComlinkListLiteral') {
+      items.push(parseListLiteral(item));
+    } else {
+      items.push(parseObjectLiteral(item));
+    }
+  }
```

Comment:
I think you can `map` over `node.items` and run `parseLiteralExample` on each item directly instead of duplicating this logic based on the node kind. Or am I missing something?